### ISSUE TITLE
IdsButton: In Angular the buttons may not be rendered as icon buttons

### ISSUE THREAD

**Issue by tmcconechy** [infor-design/enterprise-wc#1012]
**Describe the bug**
In an angular project the buttons seem not rounded when they should be. Assume that https://github.com/infor-design/enterprise-wc/blob/main/src/components/ids-button/ids-button.ts#L91 is firing before the child contents are loaded so the class is not getting added.

**To Reproduce**

1. Go to https://stackblitz.com/edit/angular-ivy-o8lcre?file=src%2Fapp%2Faccess-controls%2Fcomponents%2Faccess-list%2Fheader%2Faccess-list-header.component.html,package.json
2. Notice the icon buttons on the card are not rounded

**Expected behavior**
Should be rounded.

**Suggested Actions**
Make `type="icon"` a new button type - or a new setting (need to consider all the different settings)

**Comment by EdwardCoyle** [comment]
This may be related to #941

**Comment by EdwardCoyle** [comment]
Noting that @tmcconechy's hunch was right and the component is checking for child contents too early.

I discovered as a workaround that using the IdsButton's `icon` attribute instead of composing the icon HTML works in Angular and gets the desired outcome:

before:
```html
<ids-menu-button id="menu-button" [attr.menu]="'menu'">
  <ids-icon icon="sort-down"></ids-icon>
  <span class="audible">Sort</span>
</ids-menu-button>
``` 

after:
```html
<ids-menu-button id="menu-button" [attr.menu]="'menu'" icon="sort-down">
  <span class="audible">Sort</span>
</ids-menu-button>
```

We can recommend the second way for now, but we should try and make both work

**Comment by EdwardCoyle** [comment]
This [StackBlitz fork](https://stackblitz.com/edit/angular-ivy-hktteg?file=src/app/access-controls/components/access-detail/access-detail.component.html) demonstrates the original example using the workaround on all buttons.

**Comment by danielortiz1982** [comment]
@EdwardCoyle after viewing the StackBlitz I noticed the dropdown is being covered. Not sure if this issue is related or something new. Just wanted to point that out. 👍 
<img width="538" alt="Screen Shot 2023-01-10 at 12 44 22 PM" src="https://user-images.githubusercontent.com/5215954/211625300-f5c3c284-34f8-41a6-9b92-2183d7308f4a.png">

**Comment by EdwardCoyle** [comment]
Good catch, I saw that too.  I had tested the Stackblitz project on local against a more recent cut of WC and I wasn't seeing this issue.  It looks like IdsCard used to have a `contain: content;` rule that was [removed more recently](https://github.com/infor-design/enterprise-wc/commit/98b48f26490851003df23f4c05bbde3864b6b856#diff-29f54cb014819991cc44a9f33535fb2c07bbc378ba48661dec8e3ba9ef98baf3L13) than when we published the last beta.  If we build another beta and publish to npm, I can switch in Stackblitz so this doesn't happen.  However, I'm not sure I can fix at the moment

**Comment by tmcconechy** [comment]
yes i removed the "contain" on man for this exact issue. so should be ok now. Dont think we really need to fix his stackblitz. should be ok.

**Comment by EdwardCoyle** [comment]
Going forward using the `icon` attribute should be the recommended way to create standalone icon buttons, since the attributes run the internal lifecycle updates more reliably in Angular.  I pushed a PR to change this in the docs
